src/check.h.in

```diff
@@ -670,18 +670,30 @@ CK_DLL_EXP void CK_EXPORT _ck_assert_failed(const char *file, int line,
 #define ck_assert_uint_ge(X, Y) _ck_assert_uint(X, >=, Y)
 
 /* String comparison macros with improved output compared to ck_assert() */
-/* OP might be any operator that can be used in '0 OP strcmp(X,Y)' comparison */
-/* The x and y parameter swap in strcmp() is needed to handle >, >=, <, <= operators */
-#define _ck_assert_str(X, OP, Y) do { \
+/* OP might be any operator that can be used in '0 OP strcmp(X,Y)' comparison. */
+/* String pointer could be compared againts NULL with == and != operators. */
+/* The x and y parameter swap in strcmp() is needed to handle >, >=, <, <= operators. */
+/* If the x or y parameter is NULL its value will be printed without quotes. */
+#define _ck_assert_str(X, OP, Y, NULLEQ, NULLNE) do { \
   const char* _ck_x = (X); \
   const char* _ck_y = (Y); \
-  ck_assert_msg(0 OP strcmp(_ck_y, _ck_x), \
-    "Assertion '%s' failed: %s == \"%s\", %s == \"%s\"", #X" "#OP" "#Y, #X, _ck_x, #Y, _ck_y); \
+  const char* _ck_x_q = (_ck_x != NULL) ? "\"" : ""; \
+  const char* _ck_y_q = (_ck_y != NULL) ? "\"" : ""; \
+  ck_assert_msg( \
+    (NULLEQ && (_ck_x == NULL) && (_ck_y == NULL)) || \
+    (NULLNE && ((_ck_x == NULL) || (_ck_y == NULL)) && (_ck_x != _ck_y)) || \
+    ((_ck_x != NULL) && (_ck_y != NULL) && (0 OP strcmp(_ck_y, _ck_x))), \
+    "Assertion '%s' failed: %s == %s%s%s, %s == %s%s%s", \
+    #X" "#OP" "#Y, \
+    #X, _ck_x_q, _ck_x, _ck_x_q, \
+    #Y, _ck_y_q, _ck_y, _ck_y_q); \
 } while (0)
+
 /**
  * Check two strings to determine if 0==strcmp(X,Y)
  *
- * If not 0==strcmp(X,Y), the test fails.
+ * If ((X==NULL)||(Y==NULL))&&(X!=Y), the test fails.
```

Comment:
Although this is correct, it may be a bit to digest. Would the following be acceptable:
```
If both X and Y are NULL the test passed. However, if only one is NULL the test fails.
```